spot-client/src/common/app-state/remote-control-service/actions.js

```diff
@@ -79,3 +104,61 @@ export function setVideoMute(mute) {
                 requestTypes.VIDEO_MUTE, requestStates.ERROR)));
     };
 }
+
+/**
+ * Start wireless screensharing.
+ *
+ * @returns {Function}
+ */
+export function setWirelessScreensharing() {
+    return dispatch => {
+        dispatch(setRequestState(
+            requestTypes.SCREENSHARE,
+            requestStates.PENDING,
+            'proxy'
+        ));
+
+        return remoteControlService.setWirelessScreensharing(true, {
+            onClose: () => dispatch(setLocalWirelessScreensharing(false))
+        })
+            .then(() => {
+                dispatch(setLocalWirelessScreensharing(true));
+                dispatch(setSpotTVState({ screensharingType: 'proxy' }));
+                dispatch(setRequestState(
```

Comment:
It feels like the logic for doing the request state could be abstracted and re-used across audio/video muted and the screensharing (I still need to check how the code looks there as I don't remember). It could be taking a promise, setting PENDING when called, DONE on resolve and ERROR on reject.

Reply:
Yes, I agree. I wasn't sure if I should do it for this changeset, because another change needed is having other calls to start/stop screenshare use these actions. Do you feel it's worthwhile to make at least the action-factory change now? I will do it now or soon, so I'm okay with doing it now.

Reply:
It can come in a follow up PR.